homeassistant/components/apiai.py

```diff
@@ -0,0 +1,159 @@
+        CONF_INTENTS: {
+            cv.string: {
+                vol.Optional(CONF_ACTION): cv.SCRIPT_SCHEMA,
+                vol.Optional(CONF_SPEECH): cv.template
+            }
+        }
+    }
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Activate API.AI component."""
+    intents = config[DOMAIN].get(CONF_INTENTS, {})
+
+    hass.http.register_view(ApiaiIntentsView(hass, intents))
+
+    return True
+
+
+class ApiaiIntentsView(HomeAssistantView):
+    """Handle API.AI requests."""
+
+    url = INTENTS_API_ENDPOINT
+    name = 'api:apiai'
+
+    def __init__(self, hass, intents):
+        """Initialize API.AI view."""
+        super().__init__()
+
+        intents = copy.deepcopy(intents)
+        template.attach(hass, intents)
+
+        for name, intent in intents.items():
+            if CONF_ACTION in intent:
+                intent[CONF_ACTION] = script.Script(
+                    hass, intent[CONF_ACTION], "Apiai intent {}".format(name))
+
+        self.intents = intents
+
+    @asyncio.coroutine
+    def post(self, request):
+        """Handle API.AI."""
+        data = yield from request.json()
+
+        _LOGGER.debug('Received Apiai request: %s', data)
+
+        req = data.get('result')
+
+        if req is None:
+            _LOGGER.error('Received invalid data from Apiai: %s', data)
+            return self.json_message('Expected result value not received',
+                                     HTTP_BAD_REQUEST)
+
+        action_incomplete = req['actionIncomplete']
+
+        if action_incomplete:
+            return None
+
+        # use intent to no mix HASS actions with this parameter
+        intent = req.get('action')
+        parameters = req.get('parameters')
+        # contexts = req.get('contexts')
+        response = ApiaiResponse(parameters)
+
+        # Default Welcome Intent
+        # Maybe is better to handle this in api.ai directly?
+        #
+        # if intent == 'input.welcome':
+        #     response.add_speech(
+        #     "Hello, and welcome to the future. How may I help?")
+        #     return self.json(response)
+
+        if intent == "":
+            _LOGGER.warning('Received intent with empty action')
+            response.add_speech(
+                "You have not defined an action in your api.ai intent.")
+            return self.json(response)
+
+        config = self.intents.get(intent)
+
+        if config is None:
+            _LOGGER.warning('Received unknown intent %s', intent)
+            response.add_speech(
+                "Intent '%s' is not yet configured within Home Assistant." %
+                intent)
+            return self.json(response)
+
+        speech = config.get(CONF_SPEECH)
+        action = config.get(CONF_ACTION)
+
+        if action is not None:
+            # We can wait for the action to be executed
+            # API.AI expects a response in less than 5s
+            asyncio.ensure_future(action.async_run(response.parameters))
```

Comment:
Use `yield from action.async_run(response.parameters))`. Just like Alexa has from which you copied it ;-) https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/alexa.py#L182